From 08e870212d500887b511c3bc6b262cebf19d1af3 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Tue, 3 Dec 2019 19:32:23 +0800
Subject: [PATCH 55/62] LF-335 drm/bridge: sec-dsim: solve fallthrough build
 warnings
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

There are some fallthrough build warnings reported by
the GCC with -Wimplicit-fallthrough option like below:

drivers/gpu/drm/bridge/sec-dsim.c: In function ‘sec_mipi_dsim_write_pl_to_sfr_fifo’:
drivers/gpu/drm/bridge/sec-dsim.c:606:11: warning: this statement may fall through [-Wimplicit-fallthrough=]
  606 |   pl_data |= ((u8 *)payload)[2] << 16;
      |   ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
drivers/gpu/drm/bridge/sec-dsim.c:607:2: note: here
  607 |  case 2:
      |  ^~~~
drivers/gpu/drm/bridge/sec-dsim.c:608:11: warning: this statement may fall through [-Wimplicit-fallthrough=]
  608 |   pl_data |= ((u8 *)payload)[1] << 8;
      |   ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
drivers/gpu/drm/bridge/sec-dsim.c:609:2: note: here
  609 |  case 1:
      |  ^~~~
drivers/gpu/drm/bridge/sec-dsim.c: In function ‘sec_mipi_dsim_read_pl_from_sfr_fifo’:
drivers/gpu/drm/bridge/sec-dsim.c:687:24: warning: this statement may fall through [-Wimplicit-fallthrough=]
  687 |     ((u8 *)payload)[2] = (pl >> 16) & 0xff;
      |     ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
drivers/gpu/drm/bridge/sec-dsim.c:688:4: note: here
  688 |    case 2:
      |    ^~~~
drivers/gpu/drm/bridge/sec-dsim.c:689:24: warning: this statement may fall through [-Wimplicit-fallthrough=]
  689 |     ((u8 *)payload)[1] = (pl >> 8) & 0xff;
      |     ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
drivers/gpu/drm/bridge/sec-dsim.c:690:4: note: here
  690 |    case 1:
      |    ^~~~

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Reviewed-by: Haibo Chen <haibo.chen@nxp.com>
---
 drivers/gpu/drm/bridge/sec-dsim.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/bridge/sec-dsim.c b/drivers/gpu/drm/bridge/sec-dsim.c
index ee270a423841..8af5b7787999 100644
--- a/drivers/gpu/drm/bridge/sec-dsim.c
+++ b/drivers/gpu/drm/bridge/sec-dsim.c
@@ -603,8 +603,10 @@ static void sec_mipi_dsim_write_pl_to_sfr_fifo(struct sec_mipi_dsim *dsim,
 	switch (length) {
 	case 3:
 		pl_data |= ((u8 *)payload)[2] << 16;
+		/* fall through */
 	case 2:
 		pl_data |= ((u8 *)payload)[1] << 8;
+		/* fall through */
 	case 1:
 		pl_data |= ((u8 *)payload)[0];
 		dsim_write(dsim, pl_data, DSIM_PAYLOAD);
@@ -684,8 +686,10 @@ static int sec_mipi_dsim_read_pl_from_sfr_fifo(struct sec_mipi_dsim *dsim,
 			switch (word_count) {
 			case 3:
 				((u8 *)payload)[2] = (pl >> 16) & 0xff;
+				/* fall through */
 			case 2:
 				((u8 *)payload)[1] = (pl >> 8) & 0xff;
+				/* fall through */
 			case 1:
 				((u8 *)payload)[0] = pl & 0xff;
 				break;
-- 
2.17.1

